storage-access.bs

```diff
@@ -173,6 +178,23 @@ Modify the definition of [=source snapshot params=] in the following manner:
 1. Add a new member called <dfn for="source snapshot params">has storage access</dfn> of type [=boolean=].
 1. Add a new member called <dfn for="source snapshot params">environment id</dfn> of type opaque [=string=].
 
+A <dfn>storage access eligibility</dfn> is one of "<dfn for="storage access eligibility">unset</dfn>", "<dfn for="storage access eligibility">ineligible</dfn>", or "<dfn for="storage access eligibility">eligible</dfn>".
+
+A [=request=] has a [=storage access eligibility=] <dfn for="request">eligible for storage-access</dfn>. It is initially "<code>[=storage access eligibility/unset=]</code>".
+
+<div algorithm>
+To <dfn>determine the initial storage-access eligibility</dfn>, given a [=request=] |request|, run the following steps:
+
+    1. If |request|'s [=request/client=] is null, return "<code>[=storage access eligibility/unset=]</code>".
+    1. If |request|'s [=request/client=]'s [=environment/ancestry=] is not "<code>cross-site</code>", return "<code>[=storage access eligibility/unset=]</code>"
+    1. If |request|'s [=request/client=]'s [=environment/has storage access=] is false, return "<code>[=storage access eligibility/ineligible=]</code>".
+    1. If |request|'s [=url/origin=] is not [=/same site=] with |request|'s [=request/url=]'s [=url/origin=], return "<code>[=storage access eligibility/ineligible=]</code>".
```

Comment:
Hmm I'm not sure I understand what purpose this step serves now, would you mind elaborating it for my benefit? :)

Reply:
This checks that the request's initiator is same-site with the URL that we're considering sending cookies to. If they're cross-site, we shouldn't include cookies, because doing so would allow an attacker to manipulate an iframe into CSRF-ing itself, IIUC.

Note that this is one of the two things that change to be 'same-origin' in https://github.com/privacycg/storage-access/pull/213/commits/6b6ccef800534782bababa4b217c29abfe603277, since the rest of the web platform uses origin as the security boundary, rather than site.

Reply:
Yeah, see step 8 of https://privacycg.github.io/storage-access-headers/#http-storage-access-retry-fetch. 

I tweaked the field name (`eligible to storage-access` -> `storage access eligibility`) when I modified its type (bool -> tri-state enum) in this PR, but the SAH spec can be updated to use the new name once this PR lands.